### DISCUSSION TITLE
List albums / track by format and or bitrate

### DISCUSSION THREAD

**Discussion by catap** [beetbox/beets#4053] [Q&A] [answered]
Let assume that I have huge library where half of them in lossless like flac.

How can I find everything which is in mp3, ogg, m4a and similar formats?

**Comment by wisp3rwind** [comment]
I don't think there's a direct query to do so, but a regex query on the `format` field should work (untested!):
```
beet list "format::(mp3|m4a|...)"
```

You could use 
```
beet list -f '$format' | sort | uniq
```
to figure out which formats to include in the regex first.

Maybe a [range query](https://beets.readthedocs.io/en/stable/reference/query.html#numeric-range-queries) on the bitrate field is useful, too.

**Reply by catap** [reply]
@wisp3rwind maybe you now a way to exclude all `flac` files? :)

**Comment by wisp3rwind** [comment]
Invert the query: https://beets.readthedocs.io/en/stable/reference/query.html#query-term-negation

**Reply by catap** [reply]
BTW `-format:mp3` doesn't work but `^format:mp3` works.

**Reply by wisp3rwind** [reply]
That's probably due to ambiguities when parsing the command line, which is described in the docs at the link.

**Comment by catap** [comment]
Anyway, this is still quite tricky.

Maybe it is possible to aggregate author and/or almbums by used format and bitrate?

**Reply by wisp3rwind** [reply]
> Maybe it is possible to aggregate author and/or almbums by used format and bitrate?

Pretty sure that this can't be done right now, but sounds like a nice idea for a new plugin :)

**Reply by catap** [reply]
@sampsyo shall I create a formal Issue with feature request?

**Reply by sampsyo** [reply]
Before filing a feature request, I think it would be great to discuss a little more in this thread exactly what that feature would do. It's not yet clear to me exactly what "aggregate" means in this context.

**Reply by catap** [reply, marked as answer]
After a thinking a while I see that it is one time features, and I can't make it in generic way.
